sale_loyalty_auto_refresh/models/sale_order.py

```diff
@@ -40,10 +44,22 @@ def write(self, vals):
         return res
 
     def _auto_refresh_coupons(self):
-        orders = self.filtered(type(self)._allow_recompute_coupon_lines)
-        if orders:
-            orders = orders.with_context(skip_auto_refresh_coupons=True)
-            orders.recompute_coupon_lines()
+        self_ctx = self.with_context(skip_auto_refresh_coupons=True)
+        for order in self_ctx:
+            if order._allow_recompute_coupon_lines():
+                order._update_programs_and_rewards()
+                order.action_apply_rewards()
```

Comment:
There's no real need to change the previous structure. Yo can:
```suggestion
        orders = self.with_context(skip_auto_refresh_coupons=True).filtered(type(self)._allow_recompute_coupon_lines)
        for order in orders:
            order._update_programs_and_rewards()
            order.action_apply_rewards()